api/test/wfl/integration/sinks/datarepo_sink_test.clj

```diff
@@ -0,0 +1,131 @@
+(ns wfl.integration.sinks.datarepo-sink-test
+  "Test validation and operations on Sink stage implementations."
+  (:require [clojure.test         :refer [deftest is use-fixtures]]
+            [wfl.environment      :as env]
+            [wfl.service.postgres :as postgres]
+            [wfl.sink             :as sink]
+            [wfl.stage            :as stage]
+            [wfl.tools.datasets   :as datasets]
+            [wfl.tools.fixtures   :as fixtures]
+            [wfl.tools.queues     :refer [make-queue-from-list]]
+            [wfl.tools.resources  :as resources]
+            [wfl.tools.workloads  :refer [evalT]]
+            [wfl.util             :as util])
+  (:import [java.util UUID]
+           [wfl.util UserException]))
+
+(def ^:private ^:dynamic *dataset*)
+
+(use-fixtures :once
+  fixtures/temporary-postgresql-database
+  (fixtures/bind-fixture
+   *dataset*
+   fixtures/with-temporary-dataset
+   (datasets/unique-dataset-request
+    (env/getenv "WFL_TDR_DEFAULT_PROFILE") "testing-dataset.json")))
+
+(def ^:private outputs
+  {:outbool   true
+   :outfile   "gs://broad-gotc-dev-wfl-ptc-test-inputs/external-reprocessing/exome/develop/not-a-real.unmapped.bam"
+   :outfloat  (* 4 (Math/atan 1))
+   :outint    27
+   :outstring "Hello, World!"})
+
+;; Must be a function to capture value of *dataset* after setting up test
+;; fixtures.
+(defn ^:private datarepo-sink-request
+  "Return a valid Tera DataRepo sink request."
+  []
+  {:name        @#'sink/datarepo-sink-name
```

Comment:
Is the `@#'` prefix how we reference a `^:const`?

Reply:
This works around the fact that `datarepo-sink-name` is private - `#` creates a `Var` from the `Symbol` `sink/datarepo-sink-name` and `@` dereferences it.

Reply:
Ahhh cool, this is helpful to me… I ran into an issue trying to access a private var in test recently.  Worked around by making it public, but now I know.

Reply:
I'm not sure which is the best way to do this. I'm kind of against accessing private stuff but it's also easy...